blobfuse/include/BlobStreamer.h

```diff
@@ -87,24 +87,18 @@ class StreamObject {
         }
 
         int IncRefCount() {
-            std::lock_guard<std::mutex> lock(m_mutex);
+            //std::lock_guard<std::mutex> lock(m_mutex);
             ref_count++;
             return ref_count;
         }
 
         int DecRefCount() {
-            std::lock_guard<std::mutex> lock(m_mutex);
+            //std::lock_guard<std::mutex> lock(m_mutex);
             ref_count--;
-
-            if (ref_count == 0) {
-                Cleanup();
```

Comment:
Was this call to Cleanup unnecessary cause we have code that handles it in BlobStreamer.cpp line 214?

Reply:
It was redundant, in closeFile flow we were decrementing the count and also it was checking if its 0 then cleanup. Hence removed a implicit cleanup.